internal/integrations/utils.go

```diff
@@ -0,0 +1,194 @@
+	if err != nil {
+		return nil, fmt.Errorf("unmashalling %s, %w", path, err)
+	}
+
+	return bytes, nil
+}
+
+func populateV4Config(v3Definition Plugin, v3Configuration PluginInstanceWrapper) (*v4, error) {
+	if v3Configuration.IntegrationName != v3Definition.Name {
+		return nil, fmt.Errorf("IntegrationName != Name: %s!=%s", v3Configuration.IntegrationName, v3Definition.Name)
+	}
+
+	// The field os does not have currently a simple way to be migrated
+	if v3Definition.OS != "" {
+		log.Debugf("The old definitions had a os directive, %s. Usually it is not needed, use `when` field otherwhise", v3Definition.OS)
+	}
+
+	v4Config := &v4{}
+	for commandName, pluginV1Command := range v3Definition.Commands {
+		for _, pluginV1Instance := range v3Configuration.Instances {
+			if commandName == pluginV1Instance.Command {
+				integrationInstance := populateConfigEntry(pluginV1Instance, pluginV1Command)
+				v4Config.Integrations = append(v4Config.Integrations, integrationInstance)
+			}
+		}
+	}
+
+	return v4Config, nil
+}
+
+func populateConfigEntry(pluginV1Instance *PluginV1Instance, pluginV1Command *PluginV1Command) ConfigEntry {
+	configEntry := ConfigEntry{}
+	if len(pluginV1Command.Command) == 0 {
+		return configEntry
+	}
+
+	executable := pluginV1Command.Command[0]
+	binaryName := filepath.Base(executable)
+	configEntry.InstanceName = strings.TrimSuffix(binaryName, exeSuffix)
+	configEntry.Interval = fmt.Sprintf("%ds", pluginV1Command.Interval)
+	configEntry.Labels = pluginV1Instance.Labels
+	configEntry.User = pluginV1Instance.IntegrationUser
+	configEntry.InventorySource = pluginV1Command.Prefix
+	configEntry.Env = map[string]string{}
+	for k, v := range pluginV1Instance.Arguments {
+		configEntry.Env[strings.ToUpper(k)] = v
+	}
+
+	// Please notice that this is a simplification. If it is an absolute path we are adding it to the exec
+	// if is a relative path or a integration name, we are assuming it is a standard integration included into the path
+	if filepath.IsAbs(executable) {
+		configEntry.Exec = pluginV1Command.Command
+	} else {
+		buildCLIArgs(pluginV1Command, &configEntry)
+	}
+	return configEntry
+}
+
+func buildCLIArgs(pluginV1Command *PluginV1Command, configEntry *ConfigEntry) {
+	for index, arg := range pluginV1Command.Command {
+		if index == 0 {
+			// the first arg in command is the binary name
+			continue
+		}
+
+		sanitized := strings.TrimPrefix(arg, prefixArg)
+		sanitized = strings.TrimPrefix(sanitized, prefixArgShort)
+		if sanitized == metricArg || sanitized == inventoryArg || sanitized == eventsArg {
+			configEntry.Env[strings.ToUpper(sanitized)] = "true"
+		} else {
+			configEntry.CLIArgs = append(configEntry.CLIArgs, arg)
+		}
+	}
+}
+
+func writeOutput(pathOutput string, v4Config *v4, definitionBytes []byte, configurationBytes []byte) error {
+	if v4Config == nil {
+		return fmt.Errorf("v4Config pointer is nil")
+	}
+
+	file, err := os.OpenFile(pathOutput, os.O_RDWR|os.O_CREATE|syscall.O_TRUNC, 0666)
```

Comment:
I'd rather check for existence and rename the old file to `.old` or something along those lines. Getting data destroyed by a CLI can be frustrating :)

Reply:
I added a overwrite option, and failing in case the file does exist already. I do not like to move around user files (and it would create a loop of .old.old.old 😰 )